includes/Core/Modules/Module.php

```diff
@@ -337,11 +339,11 @@ public function is_connected() {
 			} catch ( Exception $e ) {
 				// Set every result of this batch to the exception.
 				foreach ( $results as $key => $datapoint ) {
-					if ( ! is_string( $datapoint ) || ! isset( $datapoint_services[ $datapoint ] ) || $service_identifier !== $datapoint_services[ $datapoint ] ) {
-						continue;
+					$get_service  = ! empty( $datapoint_definitions[ "GET:{$datapoint}" ] ) ? $datapoint_definitions[ "GET:{$datapoint}" ]['service'] : null;
+					$post_service = ! empty( $datapoint_definitions[ "POST:{$datapoint}" ] ) ? $datapoint_definitions[ "POST:{$datapoint}" ]['service'] : null;
+					if ( is_string( $datapoint ) && ( $service_identifier === $get_service || $service_identifier === $post_service ) ) {
```

Comment:
See above, this should be changed accordingly. Or alternatively, if I'm not mistaken, the checks here other than `is_string` may be completely unnecessary? Since we already check further above whether there is a service for this datapoint and cancel otherwise?

Reply:
Yes, updated using approach defined in the previous comment.